RE-438: :bug: Load supported type inline on iOS

scthi opened the pull request:
With the latest changes on the Basic-Auth Handling, those
types (pdfs, ...) were opened in Safari.

scthi committed:
RE-438: :bug: Load supported type inline on iOS

With the latest changes on the Basic-Auth Handling, those
types (pdfs, ...) were opened in Safari.

hassaanz commented:
What's the current status on this bug fix? Are there some known issues which still need to be fixed?

scthi commented:
> What's the current status on this bug fix? Are there some known issues which still need to be fixed?

@hassaanz We identified two issues:
1. Export of results in forms plugin. Without "basicAuth" the data is rendered inline, but with it Mobile Safari opens.
2. Link to https://marvel.com sometimes result in a blank page, because https://tpc.googlesyndication.com/safeframe/1-0-32/html/container.html is loaded by webview... (but it's only used as meta prefetch on the page...)

scthi committed:
RE-438: :bug: Check mainDocumentURL to fix the marvel.com bug

Pages referred as <meta prefetch=""> show up in the should load
but need to be ignored.

scthi committed:
RE-438: :bug: Retry first before open failed url by system

scthi commented:
I made 2 bugfixes (both **iOS only**):

* 44d5c44 fixes issue when linking to a webpage, which uses <meta prefetch="something"> tags. example: marvel.com was "sometimes" blank because the "something" page from prefetch was loaded  as "main" url
* 4a1f934 fixes issue, that results from Forms Plugin were not shown inline, but opened the Mobile Safari. example: https://de-plugin-form.eyo.net/app/download.php?eyoFormID=5ca47bd3dd8e762f01a18284&pid=5ca47bd3dd8e762f01a18284&sessionID=7217a01fb13916e614894e6f86bcfbcd&version=1

Please review again.
